### DISCUSSION TITLE
Making the vega-lite example gallery polyglot ?

### DISCUSSION THREAD

**Discussion by behrica** [vegawidget/vegawidget#172] [General]
I started here

https://github.com/vega/vega-lite/issues/7758

a discussion to make the vega lite example gallery polyglot,
so showing the specification of all the  examples in various languages.

Maybe you are interested.

**Comment by behrica** [comment]
To bring this forward, I should create an vega lite example in various languages.
Could somebody here help me with this ?

So converting:

```json
{
  "$schema": "https://vega.github.io/schema/vega-lite/v5.json",
  "description": "A simple bar chart with embedded data.",
  "data": {
    "values": [
      {"a": "A", "b": 28}, {"a": "B", "b": 55}, {"a": "C", "b": 43},
      {"a": "D", "b": 91}, {"a": "E", "b": 81}, {"a": "F", "b": 53},
      {"a": "G", "b": 19}, {"a": "H", "b": 87}, {"a": "I", "b": 52}
    ]
  },
  "mark": "bar",
  "encoding": {
    "x": {"field": "a", "type": "nominal", "axis": {"labelAngle": 0}},
    "y": {"field": "b", "type": "quantitative"}
  }
}
``` 

into the R list expected by vegawidget ?
Thanks

**Comment by ijlyttle** [comment]
The [`as_vegaspec()`](https://vegawidget.github.io/vegawidget/reference/as_vegaspec.html) function does a lot of what you need here. You can give it:

- local path to a JSON file (as character string)
- URL to a JSON file (as character string)
- JSON (as character string)
- `list()`

Basically, if you give `as_vegaspec()` a character string, it tries to do the right thing with it.

Does this help?

**Reply by behrica** [reply]
The idea is to have R code which is ready to be copy/pasted.
I am preparing a PR and they just want to see how one vega spec would look alike in all languages.
I am completely new to vegawidget, so I would just need a one time translation of the above into R code, like:
```
spec_mtcars <-
  list(
    `$schema` = vega_schema(), # specifies Vega-Lite
    description = "An mtcars example.",
    data = list(values = mtcars),
    mark = "point",
    encoding = list(
      x = list(field = "wt", type = "quantitative"),
      y = list(field = "mpg", type = "quantitative"),
      color = list(field = "cyl", type = "nominal")
    )
  ) %>% 
  as_vegaspec()
  
```

So to have one complete show case:

![image](https://user-images.githubusercontent.com/127361/138949358-16b9d388-8db5-4867-bb9e-6d9dbf6772c5.png)
